zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1010,9 +1011,11 @@ private void validateCreateRequest(String path, CreateMode createMode, Request r
             if (id == null || id.getScheme() == null) {
                 throw new KeeperException.InvalidACLException(path);
             }
-            if (id.getScheme().equals("world") && id.getId().equals("anyone")) {
+            if (id.getScheme().equals("world") && id.getId().equals("anyone") && !QuorumPeerConfig
```

Comment:
Given we force user to go "auth" option when the config is on, do we need to check if authInfo == null at line 1042 in PrepRequestProcessor.java?

Reply:
authInfo cannot be null since it comes from connection metadata.